IPython/core/history.py

```diff
@@ -554,7 +555,14 @@ def __init__(self, shell=None, config=None, **traits):
 
         if self.enabled and self.hist_file != ':memory:':
             self.save_thread = HistorySavingThread(self)
-            self.save_thread.start()
+            try:
+                self.save_thread.start()
+            except RuntimeError:
+                self.log.error(
+                    "Failed to start history saving thread. History will not be saved.",
+                    exc_info=True,
+                )
+                self.hist_file = ":memory:"
```

Comment:
+1. Should we _also_ have a condition that check if we are im emscripten to be explicit ?

Reply:
At the moment, I have to say that I don't have a clear picture of what codepaths in IPython have a chance to be useful within emscripten (cf. https://github.com/pyodide/pyodide/pull/4452#discussion_r1475158851). So I'd be reluctant to propose specific checks for the emscripten platform in places that do not already have platform-specific code (in contrast to #14316, thanks for merging that!).